src/main/java/com/strandls/dataTable/service/impl/DataTableServiceImpl.java

```diff
@@ -242,4 +301,38 @@ private DataTable dataTableSerilizer(DataTableWkt dt, DataTable prevState) {
 
 	}
 
+	@Override
+	public Activity addDataTableComment(HttpServletRequest request, CommentLoggingData comment) {
+		try {
+			comment.setMailData(generateMailData(request,comment.getRootHolderId()));
```

Comment:
pls check for null pointer before u set the mail data to the comment pojo,Also see to that API wiring is done accordingly in UI as well i.e datatble comment on ui should be calling this API.